core/src/vaults.ts

```diff
@@ -0,0 +1,361 @@
+    const contract = await getContract(network, 'CDP_MANAGER');
+    const address = await contract.urns(id);
+    if (address === '0x0000000000000000000000000000000000000000') {
+        throw new Error('Vault does not exist');
+    }
+    const collateralTypeHex = await contract.ilks(id);
+    const collateralType = ethers.utils.parseBytes32String(collateralTypeHex);
+    return {
+        id,
+        address,
+        collateralType,
+        lastSyncedAt: new Date(),
+        network,
+    };
+};
+
+export const fetchVaultBase = memoizee(_fetchVaultBase, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 2,
+});
+
+export const fetchVaultsCount = async (network: string): Promise<number> => {
+    const contract = await getContract(network, 'CDP_MANAGER');
+    const countHex = contract.count();
+    return parseInt(countHex._hex, 16);
+};
+
+const _fetchVaultCollateralParameters = async (
+    network: string,
+    collateralType: CollateralType
+): Promise<VaultCollateralParameters> => {
+    const contract = await getContract(network, 'MCD_VAT');
+    const typeHex = ethers.utils.formatBytes32String(collateralType);
+    const { rate, spot } = await contract.ilks(typeHex);
+    return {
+        stabilityFeeRate: new BigNumber(rate._hex).shiftedBy(-RAY_NUMBER_OF_DIGITS),
+        minUnitPrice: new BigNumber(spot._hex).shiftedBy(-RAY_NUMBER_OF_DIGITS),
+    };
+};
+
+export const fetchVaultCollateralParameters = memoizee(_fetchVaultCollateralParameters, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 2,
+});
+
+const _fetchVaultAmount = async (
+    network: string,
+    collateralType: CollateralType,
+    vaultAddress: string
+): Promise<VaultAmount> => {
+    const contract = await getContract(network, 'MCD_VAT');
+    const typeHex = ethers.utils.formatBytes32String(collateralType);
+    const { ink, art } = await contract.urns(typeHex, vaultAddress);
+    return {
+        initialDebtDai: new BigNumber(art._hex).shiftedBy(-DAI_NUMBER_OF_DIGITS),
+        collateralAmount: new BigNumber(ink._hex).shiftedBy(-WAD_NUMBER_OF_DIGITS),
+    };
+};
+
+export const fetchVaultAmount = memoizee(_fetchVaultAmount, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 3,
+});
+
+const _fetchGlobalLiquidationLimits = async (network: string) => {
+    const contract = await getContract(network, 'MCD_DOG');
+    const maximumProtocolDebtDaiHex = await contract.Hole();
+    const maximumProtocolDebtDai = new BigNumber(maximumProtocolDebtDaiHex._hex).shiftedBy(-RAD_NUMBER_OF_DIGITS);
+    const currentProtocolDebtDaiHex = await contract.Dirt();
+    const currentProtocolDebtDai = new BigNumber(currentProtocolDebtDaiHex._hex).shiftedBy(-RAD_NUMBER_OF_DIGITS);
+    return {
+        currentProtocolDebtDai,
+        maximumProtocolDebtDai,
+    };
+};
+
+export const fetchGlobalLiquidationLimits = memoizee(_fetchGlobalLiquidationLimits, {
+    maxAge: CACHE_EXPIRY_MS,
```

Comment:
- [ ] Those values can change more often than in 24 hours. I would say 60 seconds would be enough to avoid refetching of this value with every vault fetch. The only reason we have those caches is when we need to fetch this value in the array of say 100 vaults and we don't want to do 99 extra requests
- [ ] Please also check all other caches